### PR TITLE
feat: desigWallet support

### PR DESCRIPTION
**Pull request by ethanpx**
Hey Rainbowkit team,

Opening this PR as we have a closed-beta going on right now, and several of the dapps that want to participate use rainbowkit for their modal.

Please let me know if there's anything I can do to support this PR and get it merged.

### CODE REVIEW

**Comment by DanielSinclair** [comment]
Hi @thiennv-qng. The best way to unblock this for your dApp partners is usually to create an independent package or opensource the custom wallet connector. Our Wallet API is open and non-breaking, so you can distribute a connector and dApps can rely on it. We have a bit of a backlog at the moment! As far as RainbowKit inclusion, we'll need to QA the connector/wallet. Can you send over an invite? `daniel at rainbow.me`

**Comment by DanielSinclair** [comment]
It looks like there was a problem with the deployment:
```
[00:43:04.307] > example@0.0.67 build /vercel/path0/packages/example
[00:43:04.307] > next build
[00:43:04.307] 
[00:43:04.850] - info Linting and checking validity of types...
[00:43:07.172] - warn The Next.js plugin was not detected in your ESLint configuration. See https://nextjs.org/docs/basic-features/eslint#migrating-existing-config
[00:43:10.066] - info Creating an optimized production build...
[00:43:28.133] - info Compiled successfully
[00:43:28.141] - info Collecting page data...
[00:43:29.781] unhandledRejection ReferenceError: window is not defined
[00:43:29.782]     at desigWallet (/vercel/path0/packages/example/.next/server/pages/_app.js:2177:22)
[00:43:29.782]     at /vercel/path0/packages/example/.next/server/pages/_app.js:96:98 {
[00:43:29.782]   type: 'ReferenceError'
[00:43:29.782] }
[00:43:29.834] undefined
[00:43:29.835] /vercel/path0/packages/example:
[00:43:29.835]  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  example@0.0.67 build: `next build`
[00:43:29.835] Exit status 1
[00:43:29.853]  ELIFECYCLE  Command failed with exit code 1.
[00:43:29.878] Error: Command "pnpm run ci:example" exited with 1
```

File: packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.svg
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.svg:1]: Please resize to 28 x 28 and minify with svgo/svgomg

File: packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      qrCode: 'https://desig.io/download',
+      mobile: 'https://desig.io/download',
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/desig-wallet/panpgppehdchfphcigocleabcmcgfoca',
+    },
+
+    createConnector: () => {
+      const getProvider = () =>
+        typeof window !== 'undefined'
+          ? (window as any).desig?.ethereum
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.ts:33]: I'd also recommend you add support to Wagmi for your provider [here](https://github.com/wagmi-dev/references/blob/cf1b48dabbe42a9b71a4965928e5e789b6c656a0/packages/connectors/src/types.ts#L33) and [here](https://github.com/wagmi-dev/references/blob/cf1b48dabbe42a9b71a4965928e5e789b6c656a0/packages/connectors/src/utils/getInjectedName.ts#L47). We rely on Wagmi's internal wallet identifiers as a fallback

Comment (thiennv-qng) [reply packages/rainbowkit/src/wallets/walletConnectors/desigWallet/desigWallet.ts:33]: Thank you for your prompt feedback and I apologize for my delay. I have updated the logo to 28x28 and added support for Wagmi to my provider. You can check it out in my PR [here](https://github.com/wagmi-dev/references/pull/425.) Please let me know if there are any further issues. Thank you very much

**Comment by thiennv-qng** [comment]
> Hi @thiennv-qng. The best way to unblock this for your dApp partners is usually to create an independent package or opensource the custom wallet connector. Our Wallet API is open and non-breaking, so you can distribute a connector and dApps can rely on it. We have a bit of a backlog at the moment! As far as RainbowKit inclusion, we'll need to QA the connector/wallet. Can you send over an invite? `daniel@rainbow.me`

I have developed a small website to test connecting with Desig Wallet through Rainbow. You can check it out [here](https://desigconnect.onrender.com/evm.) I appreciate your support.

**Comment by DanielSinclair** [comment]
@thiennv-qng Seeing deployment errors when using this connector:
```
unhandledRejection ReferenceError: window is not defined
```

**Comment by thiennv-qng** [comment]
> @thiennv-qng Seeing deployment errors when using this connector:
> 
> ```
> unhandledRejection ReferenceError: window is not defined
> ```

Hi DanielSinclair, I have updated the order alphabetically and checked whether the `window` variable exists or not. Please help me to verify if it is working. Thank you.

**Comment by DanielSinclair** [comment]
@thiennv-qng When QAing, I was not able to detect the provider with the Chrome extension active.

**Comment by thiennv-qng** [comment]
> @thiennv-qng When QAing, I was not able to detect the provider with the Chrome extension active.

Sorry, I don't quite understand the issue you're facing. Could you provide more information about the problem you're encountering?

**Comment by DanielSinclair** [comment]
@thiennv-qng RainbowKit couldn't detect the extension when testing against this example deployment in Chrome with the extension installed: https://rainbowkit-example-cm4l4zr29-rainbowdotme.vercel.app/
RainbowKit relies on `installed` in the connector to be `true` and Wagmi to report that the injected provider to also `ready` [here](https://github.com/wagmi-dev/references/blob/849d7a5d2c630c1b55f3414cea034303e84250f9/packages/connectors/src/injected.ts#L88). This can occur if the content script for injection is slow or the provider is otherwise malformed.

**Comment by thiennv-qng** [comment]
> @thiennv-qng RainbowKit couldn't detect the extension when testing against this example deployment in Chrome with the extension installed: https://rainbowkit-example-cm4l4zr29-rainbowdotme.vercel.app/ RainbowKit relies on `installed` in the connector to be `true` and Wagmi to report that the injected provider to also `ready` [here](https://github.com/wagmi-dev/references/blob/849d7a5d2c630c1b55f3414cea034303e84250f9/packages/connectors/src/injected.ts#L88). This can occur if the content script for injection is slow or the provider is otherwise malformed.

I have rechecked it, and it works fine for me, and here is the video: 
https://github.com/rainbow-me/rainbowkit/assets/81727176/ea91ac23-db4c-4380-b32d-9cbcc5ed310f
due to technological reasons, we do not use an account to log into the wallet like a blockchain account, but instead need to create a separate wallet for each chain, and no addresses are pre-generated for a new user, so you need to create a new wallet address as shown in my video.

**Comment by thiennv-qng** [comment]
Hi @DanielSinclair, can you help me double-check or let me know what I need to do to get my pull request approved?

**Comment by DanielSinclair** [comment]
Closing in favor of #1565 with maintainer edits

**Pull request closed without merging**